### PR TITLE
Set the NU5 testnet activation network upgrade parameters

### LINKED ISSUE #2739: Upgrade ECC deps: librustzcash, orchard, halo2, jubjub, group, etc
ECC is cutting releases of these dependencies that all need to get updated together in our crates.

### LINKED ISSUE #2793: Set the NU5 testnet activation height, network protocol version, and consensus branch ID
See #1841 for details.

### Second NU5 Testnet Activation

Set :
- [ ] [NU5 consensus branch ID](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-chain/src/parameters/network_upgrade.rs#L130) to 0x37519621
- [ ] [NU5 testnet network protocol version](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/protocol/external/types.rs#L105) to 170_015
  - We might want to delete the mainnet network protocol version, update to 170_017, or add an `unreachable!`
- [ ] [CURRENT_NETWORK_PROTOCOL_VERSION](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/constants.rs#L119) to 170_015
- [ ] [NU5 testnet activation height](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-chain/src/parameters/network_upgrade.rs#L88) to 1_599_200

For confirmation, see the corresponding `zcashd` PR:
https://github.com/zcash/zcash/pull/5330/files#diff-e04cb1e9f29f76897c6b84334238f621ff45ba13cf180ad1d0d442d50ebe51f7R40

### Tasks

  - Switch to `Testnet` and remove `should_panic` from:
    - [`v5_transaction_is_accepted_after_nu5_activation`](https://github.com/ZcashFoundation/zebra/blob/0e89236405c872a923bbf52770dcaeee38db00b7/zebra-consensus/src/transaction/tests.rs#L209-L211).
    - v5_coinbase_transaction_is_accepted
    - any other functions that panic once NU5 has an activation height

### PR DESCRIPTION
**Pull request by conradoplg**
## Motivation

For Nu5 testnet activation we must set its activation height in the code and fix anything that expect it not to be set.

## Solution

Sets the activation height.

Removes `#[should_panic]` from tests that expected it not to be set. One of the tests that tested both mainnet and testnet was parametrized and split into two.

While testing I found a bug in  `zebra-script` where it did not check the return of a function, which would eventually lead to segfault if the function failed.

Closes #2739 , #2739 

## Review

This is in draft because tests are failing because `zebra-script` and failing when trying to parse transactions in some tests. I think the reason is that `zebra-script` must be updated to a version with a bumped `PROTOCOL_VERSION`.

I'm also not totally confident that I changed everything that was required.

Regardless this can be reviewed. @teor2345 and/or @dconnolly will probably want to take a look. Also pinging @jvff since he handled zebra-script before.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

**Comment by teor2345** [comment]
> This is in draft because tests are failing because `zebra-script` and failing when trying to parse transactions in some tests. I think the reason is that `zebra-script` must be updated to a version with a bumped `PROTOCOL_VERSION`.

Yeah I think this PR might be blocked by the dependency updates in #2739.

**Review by teor2345 — COMMENTED** [review]
This PR looks good to me, happy to merge once #2739 merges and the tests pass.

But I'd like another reviewer to double-check this PR.

**Review by jvff — DISMISSED** [review]

File: zebra-chain/src/parameters/network_upgrade.rs
Comment (teor2345) [inline zebra-chain/src/parameters/network_upgrade.rs]: See the corresponding `zcashd` change at:
https://github.com/zcash/librustzcash/pull/447/files#diff-390655b2b7dbbfda727fa47b40e98c43fd479427f293da891c00b708ba01c9b7R243

```suggestion
    (block::Height(1_599_200), Nu5),
```

**Comment by teor2345** [comment]
@conradoplg I've just updated ticket #2793 based on `zcashd` 4.5.1:

So in this PR, we need to set the:
- [x]  [NU5 consensus branch ID](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-chain/src/parameters/network_upgrade.rs#L130) to 0x37519621
- [x] [NU5 testnet network protocol version](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/protocol/external/types.rs#L105) to 170_015
- [x] [CURRENT_NETWORK_PROTOCOL_VERSION](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/constants.rs#L119) to 170_015
- [x] [NU5 testnet activation height](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-chain/src/parameters/network_upgrade.rs#L88) to 1_599_200

**Comment by teor2345** [comment]
>     * [NU5 testnet network protocol version](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/protocol/external/types.rs#L105) to 170_015

We might also want to delete the mainnet network protocol version, update to 170_017, or add an `unreachable!`.

I'm not quite sure what to do here, but I'd prefer to have no version, rather than having a wrong one.

**Comment by conradoplg** [comment]
> @conradoplg I've just updated ticket #2793 based on `zcashd` 4.5.1:
> 
> So in this PR, we need to set the:
> 
> * [ ]   [NU5 consensus branch ID](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-chain/src/parameters/network_upgrade.rs#L130) to 0x37519621
> * [ ]  [NU5 testnet network protocol version](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/protocol/external/types.rs#L105) to 170_015
> * [ ]  [CURRENT_NETWORK_PROTOCOL_VERSION](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/constants.rs#L119) to 170_015
> * [ ]  [NU5 testnet activation height](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-chain/src/parameters/network_upgrade.rs#L88) to 1_599_200

Thanks! I've done those in 361272146, except the height that was changed with your suggestion


> > ```
> > * [NU5 testnet network protocol version](https://github.com/ZcashFoundation/zebra/blob/1601c9fbb30ec33e03c37200390021cade68e0f7/zebra-network/src/protocol/external/types.rs#L105) to 170_015
> > ```
> 
> We might also want to delete the mainnet network protocol version, update to 170_017, or add an `unreachable!`.
> 
> I'm not quite sure what to do here, but I'd prefer to have no version, rather than having a wrong one.

Not sure too. I've changed it to an `unreachable!` but I'm a bit wary of it causing tests to fail, though we will know if it happens 

I've also updated ZIP-244 test vectors which changed with the new branch ID. It's now also failing due to `librustzcash` needing to be updated, which I'm assuming will also be done in https://github.com/zcashfoundation/zebra/issues/2739

**Review by teor2345 — DISMISSED** [review]
I have checked that the consensus changes in this PR match the `zcashd` release notes:
https://github.com/zcash/zcash/blob/master/doc/release-notes/release-notes-4.5.1.md#fixed-bugs-in-the-testnet-orchard-circuit

ZIP-252 should be updated soon:
https://github.com/ZcashFoundation/zebra/issues/2795#issuecomment-929913355

**Comment by teor2345** [comment]
Included in #2825.

**Pull request closed without merging**
